### PR TITLE
Rework PageFinder restrictions

### PR DESCRIPTION
**Pull request by jonathanhefner**
The idea behind this rework is that PageFinder should reject a page_id if and only if it does not resolve to a sub-path of content_path.  This changes the previous behavior in the following ways:
- non-ASCII characters are now allowed (e.g. "résumé" => "pages/résumé")
- page_ids with leading dots are now explicitly allowed (e.g. ".gitconfig" => "pages/.gitconfig") [NOTE: this could already be hacked into working with e.g. "_/../.file" => "pages/.file" or "subdir/.file" => "pages/subdir/.file"]
- page_ids which resolve to outside the content_path immediately error, instead of coercing to a sub-path (e.g. "../secret" => error, instead of "../secret" => "pages/secret")
- the trailing slash on content_path is now optional (previously, forgetting the trailing slash resulted in a confusing 404)

This rework also removes (obsoletes?) a subtle bug in the page_id character whitelist wherein `"+-="` was interpreted as `("+".."=")` instead of `["+", "-", "="]`.  Ironically, this bug is what allowed nested paths in page_ids to work at all.

### CODE REVIEW

File: lib/high_voltage/page_finder.rb [hunk 1]
```diff
@@ -2,16 +2,19 @@ module HighVoltage
...
     def initialize(page_id)
       @page_id = page_id
     end
 
     # Produce a template path to the page, in a format understood by
     # `render :template => find`
     def find
-      "#{content_path}#{clean_path}"
+      base = Pathname.new(content_path).cleanpath
+      path = base.join(@page_id.tr("\\", "")).cleanpath.to_s
```
Comment (dgalarza) [inline lib/high_voltage/page_finder.rb]: What's the purpose of the `tr` call here?

File: lib/high_voltage/page_finder.rb [hunk 2]
```diff
@@ -2,16 +2,19 @@ module HighVoltage
...
-
     def initialize(page_id)
       @page_id = page_id
     end
 
     # Produce a template path to the page, in a format understood by
     # `render :template => find`
     def find
-      "#{content_path}#{clean_path}"
+      base = Pathname.new(content_path).cleanpath
```
Comment (dgalarza) [inline lib/high_voltage/page_finder.rb]: Could we extract `base` as a private method?

File: lib/high_voltage/page_finder.rb (hunk 1 above)
Comment (dgalarza) [reply lib/high_voltage/page_finder.rb]: Could we reference `@page_id` via it's `attr_reader` instead?

**Comment by jonathanhefner** [comment]
@dgalarza I made some changes to address your feedback.  Let me know if you have any further concerns.

**Comment by dgalarza** [comment]
Thanks @jonathanhefner!

**Pull request merged**